src/deploy/osp_deployer/deployer.py

```diff
@@ -114,126 +260,105 @@ def deploy():
-            director_vm.upload_cloud_images()
-            director_vm.install_director()
-            if settings.node_type_data_map:
-                director_vm.create_subnet_routes_edge()
-                dir_pub_ip = settings.director_node.public_api_ip
-                dir_pw = settings.director_node.root_password
-                director_vm.wait_for_vm_to_come_up(dir_pub_ip,
-                                                   "root",
-                                                   dir_pw)
-                logger.info('Director VM routes set and VM is running')
-            # director_vm.render_and_upload_roles_data()
-            tester.verify_undercloud_installed()
+        if args.overcloud_only is False and is_deploy_edge_site is False:
+            deploy_undercloud(settings, sah_node, tester, director_vm)
             if args.undercloud_only:
                 return
-        else:
+        elif is_deploy_edge_site is False:
             logger.info("=== Skipped Director VM/Undercloud install")
-            director_vm = Director()
             logger.debug("Deleting overcloud stack")
             director_vm.delete_overcloud()
 
-        logger.info("=== Preparing the overcloud ===")
-
-        # The network-environment.yaml must be setup for use during DHCP
-        # server configuration
-        logger.info("Setting up network environment")
-        director_vm.setup_net_envt()
-        logger.info("Setting up dhcp server")
-        director_vm.configure_dhcp_server()
-        logger.info("Discovering nodes")
-        director_vm.node_discovery()
-        logger.info("Configuring iDRACs")
-        director_vm.configure_idracs()
-        logger.info("Importing nodes")
-        director_vm.import_nodes()
-        logger.info("Introspecting nodes")
-        director_vm.node_introspection()
-        logger.info("Assigning roles")
-        director_vm.update_sshd_conf()
-        director_vm.assign_node_roles()
-        director_vm.revert_sshd_conf()
-
-        logger.info("Configuring heat templates")
-        director_vm.setup_templates()
-
-        logger.info("=== Installing the overcloud ")
-        logger.debug("installing the overcloud ... this might take a while")
-        director_vm.deploy_overcloud()
-        tester.verify_overcloud_deployed()
+        _is_undercloud_failed, _error_uc = tester.verify_undercloud_installed()
+        _is_overcloud_failed, _error_oc = tester.verify_overcloud_deployed()
+        if (is_deploy_edge_site and _is_undercloud_failed):
+            _is_deploy_both = get_is_deploy_undercloud_and_overcloud()
+            if _is_deploy_both:
+                deploy_undercloud(settings, sah_node, tester, director_vm)
+                # recheck undercloud
+                _is_uc_failed, _err_uc = tester.verify_undercloud_installed()
+                if not _is_uc_failed:
+                    deploy_overcloud(director_vm)
+                else:
+                    raise _err_uc
```

Comment:
I think that s overkill, if undercloud failed,  likelihood is it ll faill for the same reason again

Reply:
This is checking that the undercloud deployment did not fail and then deploy the overcloud if it doesn't fail.  I believe we are taking out the prompt all together (see comment above) so this will be a mute point.

Reply:
Not relevant anymore as logic has been refactored.